qemu/softmmu/memory.c

```diff
@@ -196,19 +195,26 @@ MemoryRegion *memory_map_io(struct uc_struct *uc, ram_addr_t begin, size_t size,
     return mmio;
 }
 
+static void memory_region_remove_subregion(MemoryRegion *mr,
+                                 MemoryRegion *subregion)
+{
+    assert(subregion->container == mr);
+    subregion->container = NULL;
+    QTAILQ_REMOVE(&mr->subregions, subregion, subregions_link);
+}
+
 void memory_region_filter_subregions(MemoryRegion *mr, int32_t level)
 {
     MemoryRegion *subregion, *subregion_next;
-    memory_region_transaction_begin();
+ //   memory_region_transaction_begin();
     QTAILQ_FOREACH_SAFE(subregion, &mr->subregions, subregions_link, subregion_next) {
         if (subregion->priority >= level) {
-            memory_region_del_subregion(mr, subregion);
+            memory_region_remove_subregion(mr, subregion);
             subregion->destructor(subregion);
             g_free(subregion);
-            mr->uc->memory_region_update_pending = true;
         }
     }
-    memory_region_transaction_commit(mr);
+  //  memory_region_transaction_commit(mr);
```

Comment:
Is it safe to remove this commit?

Reply:
Yes. This function is called in two cases.

* unmap/free complete memory region (as part of destroy)
Then the parrent unmap will trigger the commit
* restore the memory state
Then the restore also restores the old flatview so no rebuild is nesecary

Reply:
I did a quick check, and it seems to make sense.

Could you also add these as comments here? So that I know the preconditions here if we are going to bump qemu version.